src/protocol.rs

```diff
@@ -437,6 +447,18 @@ impl Message for C2S {
                 try!(reader.read_exact(&mut [0u8; 3]));
                 Ok(C2S::CutText(try!(String::read_from(reader))))
             },
+            255 => {
+                let submessage_type = try!(reader.read_u8());
+                match submessage_type {
+                    0 => {
+                        let down = try!(reader.read_u16::<BigEndian>()) != 0;
+                        let keysym = try!(reader.read_u32::<BigEndian>());
+                        let keycode = try!(reader.read_u32::<BigEndian>());
+                        Ok(C2S::ExtendedKeyEvent { down: down, keysym: keysym, keycode: keycode })
+                    }
+                    _ => Err(Error::Unexpected("server to client QEMU submessage type"))
```

Comment:
I think it's worth adding a comment indicating the source for the extended message formats.